Bug report: an error occurred while list files using the object storage interface

**Describe the bug**
Using ChubaoFS as storage backend on Apache Flink environment, an error occurred while Flink was performing the List Files operation.

Exception respond by Apache Flink:
```
Caused by: org.apache.flink.fs.s3base.shaded.com.amazonaws.services.s3.model.AmazonS3Exception: Invalid Argument (Service: Amazon S3; Status Code: 400; Error Code: InvalidArgument; Request ID: bbfe064947f5422e8ddbf57ef76a63d8; S3 Extended Request ID: null), S3 Extended Request ID: null
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1639)
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1304)
 at org.apache.flink.fs.s3base.shaded.com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1056)
```

**Environment**
SDK version in http header traced by **ObjectNode**:
```
aws-sdk-java/1.11.271 Linux/3.10.0-693.5.2.el7.jd_3313.x86_64 Java_HotSpot(TM)_64-Bit_Server_VM/25.131-b11 java/1.8.0_131 scala/2.11.12, presto
```

**To Reproduce**
List files using object storage interface with multi-level directory as prefix.

Example:
The following file is stored on the ChubaoFS volume.
```
dir1 / dir1_1 / file_1_1_1.txt
```
Then invoke ListObjects interface with `dir1/dir1` as prefix.


**Expected behavior**
None.


**Additional context**
None.

--- a/master/server.go
+++ b/master/server.go
@@ -49,10 +49,14 @@ const (
 )
 
 var (
-	volNameRegexp *regexp.Regexp
-	ownerRegexp   *regexp.Regexp
-	useConnPool   = true //for test
-	gConfig       *clusterConfig
+	// regexps for data validation
+	volNameRegexp = regexp.MustCompile("^[a-zA-Z0-9][a-zA-Z0-9_.-]{1,61}[a-zA-Z0-9]$")
+	ownerRegexp   = regexp.MustCompile("^[A-Za-z][A-Za-z0-9_]{0,20}$")
+	akRegexp      = regexp.MustCompile("^[a-zA-Z0-9]{16}$")
+	skRegexp      = regexp.MustCompile("^[a-zA-Z0-9]{32}$")
+
+	useConnPool = true //for test
+	gConfig     *clusterConfig
 )
 
 // Server represents the server in a cluster
@@ -93,14 +97,7 @@ func (m *Server) Start(cfg *config.Config) (err error) {
 		log.LogError(errors.Stack(err))
 		return
 	}
-	volnamePattern := "^[a-zA-Z0-9_-]{3,256}$"
-	volNameRegexp, err = regexp.Compile(volnamePattern)
-	ownerPattern := "^[A-Za-z]{1,1}[A-Za-z0-9_]{0,20}$"
-	ownerRegexp, err = regexp.Compile(ownerPattern)
-	if err != nil {
-		log.LogError(err)
-		return
-	}
+
 	if m.rocksDBStore, err = raftstore.NewRocksDBStore(m.storeDir, LRUCacheSize, WriteBufferSize); err != nil {
 		return
 	}

--- a/objectnode/fs_volume.go
+++ b/objectnode/fs_volume.go
@@ -1126,15 +1126,14 @@ func (v *volume) findParentId(prefix string) (inode uint64, prefixDirs []string,
 
 	var parentId = proto.RootIno
 	for index, dir := range dirs {
-		curIno, curMode, err := v.mw.Lookup_ll(parentId, dir)
 
-		// Because lookup can only locate the dentry that exactly matches the parameters,
-		// if the last part of the path cannot be found, it does not mean that there is no
-		// matching dentry under the path prefix. In this case, the method returns success
-		// and previous match information to the caller.
-		if index+1 == len(dirs) && err == syscall.ENOENT {
-			return parentId, prefixDirs, nil
+		// Because lookup can only retrieve dentry whose name exactly matches,
+		// so do not lookup the last part.
+		if index+1 == len(dirs) {
+			break
 		}
+
+		curIno, curMode, err := v.mw.Lookup_ll(parentId, dir)
 
 		// If the part except the last part does not match exactly the same dentry, there is
 		// no path matching the path prefix. An ENOENT error is returned to the caller.
@@ -1147,8 +1146,11 @@ func (v *volume) findParentId(prefix string) (inode uint64, prefixDirs []string,
 			return 0, nil, err
 		}
 
+		// Because the file cannot have the next level members,
+		// if there is a directory in the middle of the prefix,
+		// it means that there is no file matching the prefix.
 		if !os.FileMode(curMode).IsDir() {
-			return 0, nil, err
+			return 0, nil, syscall.ENOENT
 		}
 
 		prefixDirs = append(prefixDirs, dir)

